sync-core/src/test/java/com/cloudant/sync/datastore/AttachmentTest.java

```diff
@@ -84,10 +85,18 @@ public void setBadAttachmentsTest() throws IOException, ConflictException, SQLEx
             Assert.fail("FileNotFoundException not thrown");
         } catch (FileNotFoundException fnfe) {
             // now check that things got rolled back
-            Cursor c1 = datastore.getSQLDatabase().rawQuery("select sequence from attachments;", null);
-            Assert.assertEquals("Attachments table not empty", c1.getCount(), 0);
-            Cursor c2 = datastore.getSQLDatabase().rawQuery("select sequence from revs;", null);
-            Assert.assertEquals("Revs table not empty", c2.getCount(), 1);
+            //TODO This needs to happen on the queue inside of the datastore
+            datastore.runOnDbQueue(new SQLQueueCallable<Object>() {
+                @Override
+                public Object call(SQLDatabase db) throws Exception {
+                    Cursor c1 = db.rawQuery("select sequence from attachments;", null);
+                    Assert.assertEquals("Attachments table not empty", c1.getCount(), 0);
+                    Cursor c2 = db.rawQuery("select sequence from revs;", null);
+                    Assert.assertEquals("Revs table not empty", c2.getCount(), 1);
```

Comment:
Have you checked these asserts actually cause test failure if they happen on the queue? I've had problems in the past where Asserts need to happen on the same thread as the tests, but that might have been on iOS.